### PR TITLE
Rain Water Trapping Problem in C++

### PR DESCRIPTION
**Pull request by adityaiiitL**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read all the contributing guidelines on Pull requests

### Description

In this I have solved a problem in which we have to find out the total units of water trapped in-between the buildings, we are taking input of number of test-cases and in each test-case we are taking n as an input that denotes the number of buildings and the next line is regarding taking input of heights of n buildings. I have solved this problem using an optimized method in C++ with complexity of O(n).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3955

### CODE REVIEW

File: C-Plus-Plus/cp/rain_water_trapping.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+using namespace std;
+#define li long int
+#define lli long long int
+void solve(){
+
+    // Question:   3 1 4 3 1 5 2
+    // left_arr:   3 3 4 4 4 5 5
+    // right_arr:  5 5 5 5 5 5 2
+    // water_left: 0 2 0 1 3 0 0
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/rain_water_trapping.cpp]: Remove the redundant comments.

File: C-Plus-Plus/cp/rain_water_trapping.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#include<bits/stdc++.h>
+using namespace std;
+#define li long int
+#define lli long long int
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/rain_water_trapping.cpp]: Don't use CP tricks here

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Don't use CP templates & tricks.

Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

File: C-Plus-Plus/cp/rain_water_trapping.cpp (hunk 2 above)
Comment (adityaiiitL) [reply C-Plus-Plus/cp/rain_water_trapping.cpp]: I have done the required changes sir. Please check and review.

File: C-Plus-Plus/cp/rain_water_trapping.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    ios_base::sync_with_stdio(false);
+    cin.tie(NULL);
+    // Taking input of number of test cases
+    long long int t;
+    cin>>t;
+    while(t--)
+    {    
+        // Calling void function
+        solve();
+    }
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/rain_water_trapping.cpp]: Remove the no of test case part as we are not using online judge so no need of that used main as driver code . Also write output in the same way your code is producing. use proper display message how you are reading and displaying outputs what is `n` doesn't make sense. Use cout for for displaying message.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/rain_water_trapping.cpp (hunk 3 above)
Comment (adityaiiitL) [reply C-Plus-Plus/cp/rain_water_trapping.cpp]: Sir, I have done the changes, Please review.

**Review by ankitaggarwal23 — COMMENTED** [review]
Don't push more than one code file in a single PR. Closing this PR as of now. Open the new PR if you want this to be reviewed and add the proper files.

**Pull request closed without merging**